Seeing internal error in manager when running bulk import test

I spun bulk import random walk test using commit d0d7b585ea7bace2f86d5168058aaf5f33eda69c and I am seeing the following failures. Maybe this is caused by the change #3044 that was just applied, but have not had a chance to investigate yet.

Seeing errors like the following in the manager logs.

```
2022-10-25T23:23:38,018 [thrift.ProcessFunction] ERROR: Internal error processing waitForFateOperation
java.lang.IllegalStateException: Unexpected prev end row org.apache.accumulo.core.metadata.schema.TabletMetadata@7dd30cce 1;r06311;r04d58
        at org.apache.accumulo.manager.tableOps.bulkVer2.LoadFiles.findOverlappingTablets(LoadFiles.java:379) ~[accumulo-manager-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.manager.tableOps.bulkVer2.LoadFiles.loadFiles(LoadFiles.java:343) ~[accumulo-manager-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.manager.tableOps.bulkVer2.LoadFiles.isReady(LoadFiles.java:104) ~[accumulo-manager-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.manager.tableOps.bulkVer2.LoadFiles.isReady(LoadFiles.java:80) ~[accumulo-manager-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.manager.tableOps.TraceRepo.isReady(TraceRepo.java:47) ~[accumulo-manager-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.core.fate.Fate$TransactionRunner.run(Fate.java:89) ~[accumulo-core-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.core.trace.TraceWrappedRunnable.run(TraceWrappedRunnable.java:52) ~[accumulo-core-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
        at org.apache.accumulo.core.trace.TraceWrappedRunnable.run(TraceWrappedRunnable.java:52) ~[accumulo-core-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at java.lang.Thread.run(Thread.java:829) ~[?:?]
```

Seeing errors like the following in the client random walk process.

```
2022-10-25T23:24:00,392 [randomwalk.bulk.BulkMinusOne] ERROR: org.apache.accumulo.core.client.AccumuloException: Internal error processing waitForFateOperation
org.apache.accumulo.core.client.AccumuloException: Internal error processing waitForFateOperation
        at org.apache.accumulo.core.clientImpl.TableOperationsImpl.doFateOperation(TableOperationsImpl.java:412) ~[accumulo-testing-shaded.jar:?]
        at org.apache.accumulo.core.clientImpl.TableOperationsImpl.doFateOperation(TableOperationsImpl.java:364) ~[accumulo-testing-shaded.jar:?]
        at org.apache.accumulo.core.clientImpl.TableOperationsImpl.doBulkFateOperation(TableOperationsImpl.java:350) ~[accumulo-testing-shaded.jar:?]
        at org.apache.accumulo.core.clientImpl.bulk.BulkImport.load(BulkImport.java:173) ~[accumulo-testing-shaded.jar:?]
        at org.apache.accumulo.testing.randomwalk.bulk.BulkPlusOne.bulkLoadLots(BulkPlusOne.java:96) ~[accumulo-testing-shaded.jar:?]
        at org.apache.accumulo.testing.randomwalk.bulk.BulkMinusOne.runLater(BulkMinusOne.java:32) ~[accumulo-testing-shaded.jar:?]
        at org.apache.accumulo.testing.randomwalk.bulk.BulkTest.lambda$visit$0(BulkTest.java:31) ~[accumulo-testing-shaded.jar:?]
        at org.apache.accumulo.core.trace.TraceWrappedRunnable.run(TraceWrappedRunnable.java:52) ~[accumulo-testing-shaded.jar:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
        at org.apache.accumulo.core.trace.TraceWrappedRunnable.run(TraceWrappedRunnable.java:52) ~[accumulo-testing-shaded.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
        at org.apache.accumulo.core.trace.TraceWrappedRunnable.run(TraceWrappedRunnable.java:52) ~[accumulo-testing-shaded.jar:?]
        at java.lang.Thread.run(Thread.java:829) ~[?:?]
Caused by: org.apache.thrift.TApplicationException: Internal error processing waitForFateOperation
        at org.apache.thrift.TServiceClient.receiveBase(TServiceClient.java:81) ~[accumulo-testing-shaded.jar:?]
        at org.apache.accumulo.core.manager.thrift.FateService$Client.recv_waitForFateOperation(FateService.java:166) ~[accumulo-testing-shaded.jar:?]
        at org.apache.accumulo.core.manager.thrift.FateService$Client.waitForFateOperation(FateService.java:151) ~[accumulo-testing-shaded.jar:?]
        at org.apache.accumulo.core.clientImpl.TableOperationsImpl.waitForFateOperation(TableOperationsImpl.java:311) ~[accumulo-testing-shaded.jar:?]
        at org.apache.accumulo.core.clientImpl.TableOperationsImpl.doFateOperation(TableOperationsImpl.java:380) ~[accumulo-testing-shaded.jar:?]
        ... 14 more
```

--- a/core/src/main/java/org/apache/accumulo/core/metadata/schema/TabletsMetadata.java
+++ b/core/src/main/java/org/apache/accumulo/core/metadata/schema/TabletsMetadata.java
@@ -357,8 +357,11 @@ public class TabletsMetadata implements Iterable<TabletMetadata>, AutoCloseable 
 
     @Override
     public Options overlapping(Text startRow, boolean startInclusive, Text endRow) {
-      var encRow = TabletsSection.encodeRow(tableId, startRow == null ? new Text("") : startRow);
-      this.range = new Range(encRow, startRow == null ? true : startInclusive, null, true);
+      var metaStartRow =
+          TabletsSection.encodeRow(tableId, startRow == null ? new Text("") : startRow);
+      var metaEndRow = TabletsSection.encodeRow(tableId, endRow);
+      this.range =
+          new Range(metaStartRow, startRow == null ? true : startInclusive, metaEndRow, true);
       this.endRow = endRow;
 
       return this;

--- a/server/manager/src/main/java/org/apache/accumulo/manager/tableOps/bulkVer2/LoadFiles.java
+++ b/server/manager/src/main/java/org/apache/accumulo/manager/tableOps/bulkVer2/LoadFiles.java
@@ -376,7 +376,8 @@ class LoadFiles extends ManagerRepo {
       }
 
       if (cmp != 0) {
-        throw new IllegalStateException("Unexpected prev end row " + currTablet + " " + loadRange);
+        throw new IllegalStateException(
+            "Unexpected prev end row " + currTablet.getExtent() + " " + loadRange);
       }
 
       // we have found the first tablet in the range, add it to the list

